File: ProcessMonitor.cs
using System;
using System.Diagnostics;
using System.Linq;
using System.Threading;

class ProcessMonitor
{
    // Thresholds for color coding
    static int HIGH_CPU = 50;
    static int MEDIUM_CPU = 30;
    static int HIGH_MEM = 500;  // MB
    static int MEDIUM_MEM = 300;  // MB

    static ConsoleColor GetColor(double value, int high, int medium)
    {
        if (value >= high) return ConsoleColor.Red;
        if (value >= medium) return ConsoleColor.Yellow;
        return ConsoleColor.Green;
    }

    static void MonitorSingleProcess(string processName)
    {
        while (true)
        {
            Console.Clear();
            Console.WriteLine($"Monitoring process: {processName}");
            Console.WriteLine(new string('-', 50));

            var processes = Process.GetProcessesByName(processName);
            if (processes.Length == 0)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine($"Process {processName} not found!");
                Console.ResetColor();
            }

            foreach (var process in processes)
            {
                double cpuUsage = GetCpuUsage(process);
                double memoryUsage = process.WorkingSet64 / (1024.0 * 1024.0); // Convert to MB

                Console.WriteLine($"PID: {process.Id} | {process.ProcessName}");
                Console.ForegroundColor = GetColor(cpuUsage, HIGH_CPU, MEDIUM_CPU);
                Console.WriteLine($"CPU: {cpuUsage:F2}%");
                Console.ForegroundColor = GetColor(memoryUsage, HIGH_MEM, MEDIUM_MEM);
                Console.WriteLine($"Memory: {memoryUsage:F2} MB");
                Console.ResetColor();
                Console.WriteLine(new string('-', 50));
            }

            Thread.Sleep(5000);
        }
    }

    static void MonitorAllProcesses()
    {
        while (true)
        {
            Console.Clear();
            Console.WriteLine("Monitoring ALL processes...");
            Console.WriteLine(new string('-', 70));

            var processes = Process.GetProcesses()
                .OrderByDescending(p => GetCpuUsage(p))
                .ToList();

            foreach (var process in processes)
            {
                double cpuUsage = GetCpuUsage(process);
                double memoryUsage = process.WorkingSet64 / (1024.0 * 1024.0); // Convert to MB

                Console.Write($"PID: {process.Id,-5} | {process.ProcessName,-20} | ");
                Console.ForegroundColor = GetColor(cpuUsage, HIGH_CPU, MEDIUM_CPU);
                Console.Write($"CPU: {cpuUsage:F2}% | ");
                Console.ForegroundColor = GetColor(memoryUsage, HIGH_MEM, MEDIUM_MEM);
                Console.WriteLine($"Memory: {memoryUsage:F2} MB");
                Console.ResetColor();
            }

            Thread.Sleep(5000);
        }
    }

    static double GetCpuUsage(Process process)
    {
        try
        {
            var startTime = DateTime.UtcNow;
            var startCpuUsage = process.TotalProcessorTime;
            Thread.Sleep(500);
            var endTime = DateTime.UtcNow;
            var endCpuUsage = process.TotalProcessorTime;
            double cpuUsedMs = (endCpuUsage - startCpuUsage).TotalMilliseconds;
            double totalMsPassed = (endTime - startTime).TotalMilliseconds;
            return (cpuUsedMs / totalMsPassed) * 100.0;
        }
        catch
        {
            return 0;
        }
    }

    static void Main()
    {
        Console.Write("Enter process name to monitor (or type 'all' for all processes): ");
        string processName = Console.ReadLine();

        if (processName.ToLower() == "all")
        {
            MonitorAllProcesses();
        }
        else
        {
            MonitorSingleProcess(processName);
        }
    }
}

File: ProcessMonitor.py
# Install psutil by typing:
# pip install psutil colorama

import psutil
import time
from colorama import Fore, Style, init

# Initialize colorama for colored output
init()

# Thresholds for color coding
HIGH_CPU = 50
MEDIUM_CPU = 30
HIGH_MEM = 500  # MB
MEDIUM_MEM = 300  # MB

def get_color(value, high, medium):
    """Returns color based on thresholds."""
    if value >= high:
        return Fore.RED
    elif value >= medium:
        return Fore.YELLOW
    return Fore.GREEN

def monitor_single_process(process_name):
    """Monitor a single process by name."""
    while True:
        found = False
        print(f"\nMonitoring process: {process_name}")
        print("-" * 50)

        for proc in psutil.process_iter(attrs=['pid', 'name', 'cpu_percent', 'memory_info']):
            if proc.info['name'].lower() == process_name.lower():
                found = True
                cpu = proc.info['cpu_percent']
                mem = proc.info['memory_info'].rss / (1024 * 1024)  # Convert to MB

                print(f"PID: {proc.info['pid']} | {Fore.CYAN}{proc.info['name']}{Style.RESET_ALL}")
                print(f"CPU: {get_color(cpu, HIGH_CPU, MEDIUM_CPU)}{cpu:.2f}%{Style.RESET_ALL}")
                print(f"Memory: {get_color(mem, HIGH_MEM, MEDIUM_MEM)}{mem:.2f} MB{Style.RESET_ALL}")
                print("-" * 50)

        if not found:
            print(Fore.RED + f"Process {process_name} not found!" + Style.RESET_ALL)

        time.sleep(5)

def monitor_all_processes():
    """Monitor all running processes."""
    while True:
        print("\nMonitoring ALL processes...")
        print("-" * 70)
        
        processes = sorted(psutil.process_iter(attrs=['pid', 'name', 'cpu_percent', 'memory_info']), 
                           key=lambda p: p.info['cpu_percent'], reverse=True)

        for proc in processes:
            cpu = proc.info['cpu_percent']
            mem = proc.info['memory_info'].rss / (1024 * 1024)  # Convert to MB

            print(f"PID: {proc.info['pid']:5} | {proc.info['name'][:20]:20} | "
                  f"CPU: {get_color(cpu, HIGH_CPU, MEDIUM_CPU)}{cpu:.2f}%{Style.RESET_ALL} | "
                  f"Memory: {get_color(mem, HIGH_MEM, MEDIUM_MEM)}{mem:.2f} MB{Style.RESET_ALL}")

        time.sleep(5)

if __name__ == "__main__":
    process_name = input("Enter process name to monitor (or type 'all' for all processes): ")

    if process_name.lower() == "all":
        monitor_all_processes()
    else:
        monitor_single_process(process_name)
